doc/BUILDSERVER.md

```diff
@@ -0,0 +1,98 @@
+# Buildserver
+
+## Introduction
+Most of the build jobs on the jenkins server that handles automated testing of
+contributions have been replaced by a single
+[job](https://github.com/ElektraInitiative/libelektra/).
+
+
+## Jenkinsfile
+Jenkinsfiles describe what actions the build system should execute on what
+build slave.
+Currently Elektra uses two different files.
+
+One is for daily maintanence tasks, like cleaning up build servers
+([Jenkinsfile.daily](https://github.com/ElektraInitiative/libelektra/blob/master/scripts/jenkins/Jenkinsfile.daily))
+.
+The other one is triggered on code changes and is for testing changes to the
+codebase
+([Jenkinsfile](https://github.com/ElektraInitiative/libelektra/blob/master/scripts/jenkins/Jenkinsfile))
+.
+
+The language used is a groovy based DSL described in the
+[Jenkinsfile book](https://jenkins.io/doc/book/pipeline/jenkinsfile/).
+Most groovy syntax is allowed to describe pipelines in a Jenkinsfile, but it
+is executed in a sandbox which might block certain calls.
+
+Since plugins might extend the pool of available commands or variables a full
+list of currently available syntax can be seen in
+[pipeline
+syntax](https://build.libelektra.org/jenkins/job/libelektra/pipeline-syntax/)
+after a login to the build server.
+
+## Security
+Since a malicious PR could easily destroy the build server and slaves or expose
+credentials some restrictions are introduced.
+Only PR authors that have the right to push to libelektra can modify the
+Jenkinsfile and have those changes be respected for the respective branch.
+
+This setting can be modified on the respective build job configuration site.
+
+## Triggers
+All Triggers are described in the configuration of the respective build jobs.
+
+The
+[daily build](https://build.libelektra.org/jenkins/job/elektra-jenkinsfile-daily/)
+is executed according to a cron schedule.
+Additionally it can manually triggered by its defined phrase as described in
+`GIT.md`.
+
+The [libelektra](https://build.libelektra.org/jenkins/job/libelektra/)
+build is triggered for all branches of the libelektra repository except for
+`debian`.
+Additionally all open branches in forks targeting libelektras repository via
+PRs are going to be build.
+Pushes to any of those branches will trigger a new build automatically.
+Again a trigger phrase has been added that allows forced rebuilds via the
+repository.
+
+
+## Test Environments
+We use Docker Containers to provide the various Test enviornments.
+
+They are described
+[in the repository](https://github.com/ElektraInitiative/libelektra/tree/master/scripts/docker)
+and the Jenkinsfile describes how to build them
+.
+If a rebuild of the images is needed is determined by the hash of the
```

Comment:
The daily jobs also build them?

Reply:
Not currently. I fail to see where this is implied?

Reply:
No its not implied but wouldn't make it sense?